api/jobs.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/ONSdigital/log.go/log"
+	uuid "github.com/satori/go.uuid"
+	"net/http"
+)
+
+//NewID generates a random uuid and returns it as a string.
+var NewID = func() string {
+	return uuid.NewV4().String()
+}
+
+// CreateJobHandler returns a function that generates a new Job resource containing default values in its fields.
+func (api *JobStorerAPI) CreateJobHandler(ctx context.Context) http.HandlerFunc {
+	log.Event(ctx, "Entering CreateJobHandler function, which generates a new Job resource.", log.INFO)
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+
+		id := NewID()
+
+		//Create job in job store
+		newJob, err := api.jobStore.CreateJob(ctx, id)
+		if err != nil {
+			log.Event(ctx, "creating and storing job failed", log.Error(err), log.ERROR)
```

Comment:
Does this need to set an error `http.Status` on the response?

Reply:
yes it does, thanks, now added